Plugins/AWSLambdaDeployer/Plugin.swift

```diff
@@ -0,0 +1,433 @@
+                          samDeploymentDescriptorFilePath: samDeploymentDescriptorFilePath,
+                          verboseLogging: configuration.verboseLogging)
+        
+        // deploy the functions
+        if !configuration.noDeploy {
+            try self.deploy(samExecutablePath: samExecutablePath,
+                            samDeploymentDescriptorFilePath: samDeploymentDescriptorFilePath,
+                            stackName : configuration.stackName,
+                            verboseLogging: configuration.verboseLogging)
+        }
+        
+        // list endpoints
+        if !configuration.noList {
+            let output = try self.listEndpoints(samExecutablePath: samExecutablePath,
+                                                samDeploymentDescriptorFilePath: samDeploymentDescriptorFilePath,
+                                                stackName : configuration.stackName,
+                                                verboseLogging: configuration.verboseLogging)
+            print(output)
+        }
+    }
+    
+    private func generateDeploymentDescriptor(projectDirectory: Path,
+                                              buildConfiguration: PackageManager.BuildConfiguration,
+                                              swiftExecutable: Path,
+                                              samDeploymentDescriptorFilePath: String,
+                                              archivePath: String,
+                                              force: Bool,
+                                              verboseLogging: Bool) throws {
+        print("-------------------------------------------------------------------------")
+        print("Generating SAM deployment descriptor")
+        print("-------------------------------------------------------------------------")
+        
+        //
+        // Build and run the Deploy.swift package description
+        // this generates the SAM deployment descriptor
+        //
+        let deploymentDescriptorFileName = "Deploy.swift"
+        let deploymentDescriptorFilePath = "\(projectDirectory)/\(deploymentDescriptorFileName)"
+        let sharedLibraryName = "AWSLambdaDeploymentDescriptor" // provided by the swift lambda runtime
+        
+        // Check if Deploy.swift exists. Stop when it does not exist.
+        guard FileManager.default.fileExists(atPath: deploymentDescriptorFilePath) else {
+            print("`Deploy.Swift` file not found in directory \(projectDirectory)")
+            throw DeployerPluginError.deployswiftDoesNotExist
+        }
+        
+        do {
+            let cmd = [
+                swiftExecutable.string,
+                "-L \(projectDirectory)/.build/\(buildConfiguration)/",
+                "-I \(projectDirectory)/.build/\(buildConfiguration)/",
+                "-l\(sharedLibraryName)",
+                deploymentDescriptorFilePath,
+                "--archive-path", archivePath
+            ]
```

Comment:
interesting. @neonichu wdyt?

Reply:
The reason why this exists is described here 
https://github.com/swift-server/swift-aws-lambda-runtime/pull/291#discussion_r1135571744

Reply:
I think we'd need a different way to determine the built products directory, this will e.g. not work in Xcode and also isn't guaranteed to be a stable path in SwiftPM.

Reply:
1. I think that is fine, but as written, there's no guarantee that the library is built before running the plugin or am I missing something?
2. No, there's currently no way to get that info.

Reply:
@neonichu 
I initially added the dependency on the plugin itself in the project Package.swift.  However, starting with Swift 5.8, this generates an error (plugins can not have dependencies) 

I now add the dependency to any target of the lambda function project itself. So responsibility shifted to the Lambda function developer. I will document it as a required step when creating a new project.

Can you think about another alternative ?

Reply:
@nonichu @tomerd I found a better way to make the sure the library is built before the plugin needs it. I added a step in teh plugin to compile the library.

Basically, before doing anything else, the plugin launches the command `swift build -c release --product AWSLambdaDeploymentDescriptor` that will make sure the lib...dylib file is available in the `.build` directory, before the next command that depends on it is launched.

This implementation hides the details to user and is robust. I will resolve this comment if there is no further question.